homework/index.js

```diff
@@ -29,19 +31,120 @@
     });
     return elem;
   }
-
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  const root = document.getElementById('root');
+  function singlePageApplication(arr) {
+    arr.sort((a, b) => a.name.localeCompare(b.name));
+    // header
+    const div = createAndAppend('div', root, {
+      id: 'header-div',
+    });
+    createAndAppend('h1', div, { text: 'HYF Repositories', id: 'header' });
+    const selectMenu = createAndAppend('select', div, {
+      id: 'select-menu',
+    });
+    for (let i = 0; i < arr.length; i++) {
+      createAndAppend('option', selectMenu, {
+        text: arr[i].name,
+        value: i,
+      });
+    }
+    // right and left part
+    const containerDiv = createAndAppend('div', root, {
+      id: 'container',
+      class: 'container',
+    });
+    // right part
+    const firstTime = ['name', 'description', 'forks', 'updated_at'];
+    const rightDiv = createAndAppend('div', containerDiv, {
+      id: 'right-div',
+      class: 'contained',
+    });
+    const tHeads = ['Repository: ', 'Description: ', 'Forks: ', 'Update: '];
+    const table = createAndAppend('table', rightDiv, {
+      id: 'tableOfInformation',
+    });
+    for (let i = 0; i < tHeads.length; i++) {
+      const tr = createAndAppend('tr', table, {});
+      createAndAppend('th', tr, {
+        text: tHeads[i],
+      });
+      if (i === 0) {
```

Comment:
Why 0? 
every hard coded number like this is called a magic number, and magic numbers aren't good,
let's say it's Friday and you get back home after a long work day you won't remember why it was a 0 not may be 3 or 5 on Monday!

what about: `tHeads[i] === 'Repository: '` ?
or you may have a variable with the name of that magic number! But it have to be all `CAPS` because it's constant.